File: network-equipment-pms/pms-project/src/main/java/com/dp/plat/project/entity/ProjectMember.java
package com.dp.plat.project.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.dp.plat.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

/**
 * 项目成员
 * 关联表：pms_project_member
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("pms_project_member")
public class ProjectMember extends BaseEntity {

    private static final long serialVersionUID = 1L;

    private Long projectId;
    private Long userId;
    private String userName;

    /** PROJECT_MANAGER / PROJECT_MEMBER / APPROVER / VIEWER / CUSTOMER */
    private String role;

    private LocalDate joinDate;
    private LocalDate leaveDate;
}

File: network-equipment-pms/pms-frontend/src/api/project-member.ts
import { del, get, post, put } from '@/utils/request'

export interface ProjectMember {
  id?: number
  projectId: number
  userId: number
  userName?: string
  role: 'PROJECT_MANAGER' | 'PROJECT_MEMBER' | 'APPROVER' | 'VIEWER' | 'CUSTOMER'
  joinDate?: string
  leaveDate?: string
}

export function listMembersByProjectId(projectId: number) {
  return get<ProjectMember[]>(`/api/project/member/project/${projectId}`)
}

export function createMember(data: ProjectMember) {
  return post<ProjectMember>('/api/project/member', data)
}

export function updateMember(data: ProjectMember) {
  return put<ProjectMember>('/api/project/member', data)
}

export function deleteMember(id: number) {
  return del<void>(`/api/project/member/${id}`)
}
